pages/index.js

```diff
@@ -1,95 +1,124 @@
 import Head from "next/head";
-<<<<<<< HEAD
 import { useRouter } from "next/router";
+import { useState } from 'react';
 import { siteTitle } from "../components/Layout";
 import {
   getAuth,
-  connectAuthEmulator,
   signInWithEmailAndPassword,
-  createUserWithEmailAndPassword
+  createUserWithEmailAndPassword,
+  onAuthStateChanged
 } from "firebase/auth"
 import { firebaseApp } from "../config/firebaseConfig";
-import {
-  txtEmail,
-  txtPassword,
-  btnLogin,
-  btnSignup
-} from "../config/authUI";
-import LoginForm from "../components/login/LoginForm";
-import SignUpForm from "../components/login/SignUpForm";
-import { async } from "@firebase/util";
-
-
-const auth = getAuth(firebaseApp);
-connectAuthEmulator(auth, "http://localhost:9099");
 
-//Logging in:
 
-const loginEmailPassword = async () => {
-  const loginEmail = txtEmail.value;
-  const loginPassword = txtPassword.value;
-  try {
-    const userCredential = await signInWithEmailAndPassword(auth, loginEmail, loginPassword);
-    console.log(userCredential.user);
-  }
-  catch (error) {
-    console.log(error);
 
-  }
-}
+const Home = () => {
 
-btnLogin.addEventListener("click", loginEmailPassword);
+  const router = useRouter();
+  const auth = getAuth(firebaseApp);
 
+  const [loginEmail, setLoginEmail] = useState('');
+  const [loginPassword, setLoginPassword] = useState('');
+  const [userName, setUserName] = useState('');
+  const [signupEmail, setSignupEmail] = useState('');
+  const [signupPassword, setSignupPassword] = useState('');
+  const [signupPassConf, setSignupPassConf] = useState('');
+  const [loginNotification, setLoginNotification] = useState('');
+  const [signupNotification, setSignupNotification] = useState('');
 
-//Creating account:
+  onAuthStateChanged(auth, (user) => {
+    if (user) {
+      const uid = user.uid;
+    } else {
 
-const createAccount = async () => {
-  const loginEmail = txtEmail.value;
-  const loginPassword = txtPassword.value;
+    }
+  });
 
-  try {
-    const userCredential = await createUserWithEmailAndPassword(auth, loginEmail, loginPassword);
-    console.log(userCredential.user);
+  const handleLogin = (e) => {
+    e.preventDefault();
+    signInWithEmailAndPassword(auth, loginEmail, loginPassword)
+      .then(
+        setLoginNotification(
+          'You are logged in'
+        ),
+        setTimeout(() => {
+          setLoginNotification('')
+        }, 2000)
+      )
+      .catch((err) => {
+        console.log(err.code, err.message)
+        setLoginNotification(err.message)
+        setTimeout(() => {
+          setLoginNotification('')
+        }, 2000)
+      })
+    setLoginEmail('')
+    setLoginPassword('')
+    router.push({
+      pathname: "/[user]",
+      query: {
+        user: "user1AuthId",
```

Comment:
This line is what caused the the day before yesterday's 50,000-read  and yesterday's 16,000-read consumption of Firebase data. You have a hardcoded `user1AuthId` that had been pushed through the routes into the `Sidebar.js` and `pges/[user]/[list].js` that continued to make API calls to Firestore to fetch data, WHETHER YOUR LOGIN FUNCTIONALITY WORKED OR NOT!! You profess that you have this `router.push()` method live in there to see if you logged the user in and it'd take you to the signed in page (`pages/[user]/index.js`). But in reality, regardless if the user was signed in, this `router.push()` call with the hard-coded `user1AuthId` in there was rendering the `/pages/[user]/index.js` (and continue to consume data) as long as your `npm run dev` stayed alive while you were working whatever you thought you were working on--your auth functioning had nothing to do with this routing calls continuous running and eating up the data consumption. If the user is not signed in it run and chewed up the data consumption, and if the user is signed in it still run and chewed it up. You thought you saw the `page/[user]/index.js` page render because you signed in the user, but the fact was, that page was being rendered EVERYTHING YOU HIT SAVE and had nothing to do with your auth. Also, the` router.push()` call on line 57, if the app is ready to test the sign in routing, needs to be in the `.then()` promise call block between lines 41 and 46. The user in the Firebase tutorial made a mistake in placing the router.push("/") call outside the `.then()` block--which means that regardless the user has failed to sign in the would would still be redirected to the home page, which is wrong and makes the login process meaningless.